src/compiler.rkt

```diff
@@ -223,108 +219,106 @@
     (vector-set! vprecs (- n varc) final-parent-precision)
 
     (define new-exponents (get-exponent op output srcs))
-    
     (define child-precision (+ exps-from-above new-exponents))
-    (map (lambda (x) (when (>= x varc)  ; when tail register is not a variable
-                       (vector-set! vprecs (- x varc)
-                                    (max ; check whether this op already has a precision that is higher
-                                     (vector-ref vprecs (- x varc))
-                                     child-precision))))
-         tail-registers)))
+    (for-each (lambda (x) (when (>= x varc)  ; when tail register is not a variable
+                            (vector-set! vprecs (- x varc)
+                                         (max ; check whether this op already has a precision that is higher
+                                          (vector-ref vprecs (- x varc))
+                                          child-precision))))
+              tail-registers)))
 
 (define (get-exponent op output srcs)
   (cond
       [(equal? op ival-add)
        ; log[Г+] = max(log[x], log[y]) - log[x + y]
+       ;                               ^^^^^^^^^^^^
+       ;                               possible cancellation
```

Comment:
Interesting. Do we need to worry about this?

Reply:
I just wanted to add some comments so that later I will not get lost by looking at the formulas

Reply:
Oh, I see, I misread. You're pointing out that this is the part where cancellation is detected. Sure.